src/services/ServiceLocator.hh

```diff
@@ -0,0 +1,96 @@
+#ifndef RA_SERVICE_LOCATOR_HH
+#define RA_SERVICE_LOCATOR_HH
+#pragma once
+
+#include <assert.h>
+#include <memory>
+
+namespace ra {
+namespace services {
+
+class ServiceLocator
+{
+public:
+    /* returns a pointer to the requested service, or nullptr if not found
+     */
+    template <class TClass>
+    static const TClass* Get()
+    {
+        return Service<TClass>::Get();
+    }
+
+    /* returns a pointer to the requested service, or nullptr if not found
+     *
+     * WARNING: mutable services are not guaranteed to be thread-safe. prefer using non-mutable
+     *          services whenever possible.
+     */
+    template <class TClass>
+    static TClass* GetMutable()
+    {
+        return Service<TClass>::Get();
+    }
+    
+    /* registers a service
+     */
+    template <class TClass>
+    static void Provide(TClass* pInstance)
+    {
+        Service<TClass>::s_pInstance.reset(pInstance);
+    }
+    
+    /* temporarily overrides a service registration - primarily used in unit tests
+     */
+    template <class TClass>
+    class ServiceOverride
```

Comment:
This is meant to simplify registering Mock instances in the unit tests. It's not being used yet.